test/views/AgentForOrgCheckRegisteredDetailsViewSpec.scala

```diff
@@ -0,0 +1,94 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package views
+
+import forms.AgentForOrgCheckRegisteredDetailsFormProvider
+import models.{DraftId, NormalMode, TraderDetailsWithCountryCode}
+import play.twirl.api.HtmlFormat
+import views.behaviours.ViewBehaviours
+import views.html.AgentForOrgCheckRegisteredDetailsView
+
+class AgentForOrgCheckRegisteredDetailsViewSpec extends ViewBehaviours {
+
+  val messageKeyPrefix = "checkRegisteredDetails"
+
+  val doesNotConsentToDisclosureOfPersonalData: TraderDetailsWithCountryCode =
+    traderDetailsWithCountryCode.copy(consentToDisclosureOfPersonalData = false)
+
+  val emptyCDSFullName: TraderDetailsWithCountryCode =
+    traderDetailsWithCountryCode.copy(CDSFullName = "")
+
+  val form: AgentForOrgCheckRegisteredDetailsFormProvider =
+    app.injector.instanceOf[AgentForOrgCheckRegisteredDetailsFormProvider]
+
+  val view: AgentForOrgCheckRegisteredDetailsView = app.injector.instanceOf[AgentForOrgCheckRegisteredDetailsView]
+
+  val viewViaApply: HtmlFormat.Appendable  =
+    view(form.apply(), traderDetailsWithCountryCode, NormalMode, DraftId(1L))(fakeRequest, messages)
+  val viewViaRender: HtmlFormat.Appendable =
+    view.render(form.apply(), traderDetailsWithCountryCode, NormalMode, DraftId(1L), fakeRequest, messages)
+  val viewViaF: HtmlFormat.Appendable      =
+    view.f(form.apply(), traderDetailsWithCountryCode, NormalMode, DraftId(1L))(fakeRequest, messages)
+
+  "AgentForOrgCheckRegisteredDetailsView" - {
+    normalPage(messageKeyPrefix, "agentOrg")()
+  }
+
+  "the trader does not consent to disclosure of personal data" - {
+    val viewAlternate =
+      view(form.apply(), doesNotConsentToDisclosureOfPersonalData, NormalMode, DraftId(1L))(
+        fakeRequest,
+        messages
+      )
+
+    pageByMethodWithAssertions(
+      viewAlternate,
+      "checkRegisteredDetails.private",
+      "",
+      doesNotConsentToDisclosureOfPersonalData.EORINo
+    )() {
+      "contains the correct heading" in {
+        assertContainsMessages(
+          asDocument(viewAlternate),
+          messages("checkRegisteredDetails.private.heading", doesNotConsentToDisclosureOfPersonalData.EORINo)
+        )
+      }
+    }
+
+    "when CDSFullName is an empty string" - {
+      val viewAlternate =
+        view(form.apply(), emptyCDSFullName, NormalMode, DraftId(1L))(
+          fakeRequest,
+          messages
+        )
+
+      pageByMethodWithAssertions(
+        viewAlternate,
+        messageKeyPrefix,
+        "agentOrg",
+        emptyCDSFullName.EORINo
+      )() {
+        "the page does not contain the registered business name on the form" in {
+          assertNotContainingMessages(
+            asDocument(viewAlternate),
+            messages("checkRegisteredDetails.heading.2")
+          )
+        }
+      }
+    }
+  }
```

Comment:
Also, it would less confusing to use those methods going forward as one would need to make sure when pageByMethodWithAssertions is used, the tests are not duplicated.

Reply:
The reason I coupled them was because you can have a different heading and title on some of these pages. So i wanted to test it again, also thought if we wanted to check anything else about the structure more generally in pageByMethod, that could possibly change when the view is adding in extra components dynamically.

Reply:
Yh I noticed that, but one is also likely to duplicate tests in that way. Let's chat

Reply:
As discussed please decouple.